onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -462,22 +461,44 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<onnxruntime:
         const OrtValue* input_tensor = ort.KernelContext_GetInput(context, input_indexes[i]);
         auto tensor_info = ort.GetTensorTypeAndShape(input_tensor);
         const auto& tensor_shape = ort.GetTensorShape(tensor_info);
+        auto tensor_type = ort.GetTensorElementType(tensor_info);
         ort.ReleaseTensorTypeAndShapeInfo(tensor_info);
-        const float* input = ort.GetTensorData<float>(input_tensor);
 
         const int input_batch_size = tensor_shape[0];
         if (i > 0 && batch_size != input_batch_size) {
           ORT_THROW("Input batch size is inconsistent");
         }
         batch_size = input_batch_size;
 
-        CHECK_CUDA(cudaMalloc(&buffers[i], input_batch_size * input_dim_sizes[i] * sizeof(float)));
-        CHECK_CUDA(cudaMemcpy(buffers[i], input, input_batch_size * input_dim_sizes[i] * sizeof(float), cudaMemcpyHostToDevice));
+        int input_size = batch_size * input_dim_sizes[i];
+        if (tensor_type == ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT) {
+          const float* input = const_cast<float*>(ort.GetTensorData<float>(input_tensor));
+          CHECK_CUDA(cudaMalloc(&buffers[i], input_size * sizeof(float)));
+          CHECK_CUDA(cudaMemcpy(buffers[i], input, input_size * sizeof(float), cudaMemcpyHostToDevice));
+        } else if (tensor_type == ONNX_TENSOR_ELEMENT_DATA_TYPE_INT8) {
+          const int8_t* input = const_cast<int8_t*>(ort.GetTensorData<int8_t>(input_tensor));
+          CHECK_CUDA(cudaMalloc(&buffers[i], input_size * sizeof(int8_t)));
+          CHECK_CUDA(cudaMemcpy(buffers[i], input, input_size * sizeof(int8_t), cudaMemcpyHostToDevice));
+        } else if (tensor_type == ONNX_TENSOR_ELEMENT_DATA_TYPE_INT32) {
+          const int32_t* input = const_cast<int32_t*>(ort.GetTensorData<int32_t>(input_tensor));
+          CHECK_CUDA(cudaMalloc(&buffers[i], input_size * sizeof(int32_t)));
+          CHECK_CUDA(cudaMemcpy(buffers[i], input, input_size * sizeof(int32_t), cudaMemcpyHostToDevice));
+        } else {
+          Status(common::ONNXRUNTIME, common::FAIL, "Input tensor type " + std::to_string(tensor_type) + " is not supported.");
```

Comment:
why allocate a Status() , doesn't seem used anywhere.

Reply:
same comment for other Status() allocations.

Reply:
looks like this compute_func() always returns 0, which doens't seem correct.
if there are errors, we need to return 1.